java/ingest/ingest-runner/src/test/java/sleeper/ingest/impl/IngestCoordinatorUsingDirectWriteBackedByArrowIT.java

```diff
@@ -16,130 +16,232 @@
+        Instant stateStoreUpdateTime = Instant.parse("2023-08-08T11:20:00Z");
+        IngestCoordinatorTestParameters parameters = createTestParameterBuilder()
+                .fileNames(List.of("leftFile", "rightFile"))
+                .fileUpdatedTimes(() -> stateStoreUpdateTime)
+                .stateStore(stateStore)
+                .schema(recordListAndSchema.sleeperSchema)
+                .workingDir(ingestLocalWorkingDirectory)
+                .build();
 
-        IngestTestHelper<Record> test = test(recordListAndSchema, tree, arrow -> arrow
+        // When/Then
+        Consumer<ArrowRecordBatchFactory.Builder<Record>> arrowConfig = arrow -> arrow
                 .workingBufferAllocatorBytes(32 * 1024L)
-                .batchBufferAllocatorBytes(1024 * 1024L)
-                .maxNoOfBytesToWriteLocally(64 * 1024 * 1024L));
-        assertThatThrownBy(() ->
-                test.ingestAndVerify(keyToPartitionNoMappingFn, partitionNoToExpectedNoOfFilesMap))
+                .batchBufferAllocatorBytes(32 * 1024L)
+                .maxNoOfBytesToWriteLocally(64 * 1024 * 1024L);
+        TestIngestType ingestType = directWriteBackedByArrowWriteToLocalFile(arrowConfig);
+        assertThatThrownBy(() -> ingestRecords(recordListAndSchema, parameters, ingestType))
                 .isInstanceOf(OutOfMemoryException.class)
                 .hasNoSuppressedExceptions();
     }
 
-    private IngestTestHelper<Record> test(RecordGenerator.RecordListAndSchema recordListAndSchema,
-                                          PartitionTree tree,
-                                          Consumer<ArrowRecordBatchFactory.Builder<Record>> arrowConfig) throws Exception {
-        return IngestTestHelper.from(temporaryFolder,
-                        AWS_EXTERNAL_RESOURCE.getHadoopConfiguration(),
-                        recordListAndSchema)
-                .createStateStore(AWS_EXTERNAL_RESOURCE.getDynamoDBClient(), tree)
-                .directWrite()
-                .backedByArrow(arrowConfig);
+    private IngestCoordinatorTestParameters.Builder createTestParameterBuilder() {
+        return IngestCoordinatorTestParameters
+                .builder()
+                .temporaryFolder(temporaryFolder)
+                .hadoopConfiguration(configuration);
+    }
+
+    private static void ingestRecords(RecordGenerator.RecordListAndSchema recordListAndSchema,
+                                      IngestCoordinatorTestParameters ingestCoordinatorTestParameters,
+                                      TestIngestType ingestType) throws Exception {
+        try (IngestCoordinator<Record> ingestCoordinator =
+                     ingestType.createIngestCoordinator(ingestCoordinatorTestParameters)) {
+            for (Record record : recordListAndSchema.recordList) {
+                ingestCoordinator.write(record);
+            }
+        }
+    }
+
+    private static void assertThatOrderedRecordsInRange(List<Record> records, LongStream range) {
```

Comment:
I find it difficult to tell what this is asserting. I think the idea is to check the values for a particular field (in this case key0), and assert that they all appear in the range, and their order is the same in the list of records as it is in the range?

Maybe something like `assertThatRecordsHaveFieldValuesThatAllAppearInRangeInSameOrder(List<Record> records, String fieldName, LongStream range)`?